app/qml/inputs/MMPasswordInput.qml

```diff
@@ -8,46 +8,26 @@
  ***************************************************************************/
 
 import QtQuick
-import QtQuick.Controls
-import QtQuick.Controls.Basic
-import "../components"
 
-MMBaseInput {
-  id: root
-
-  property alias placeholderText: textField.placeholderText
-  property alias text: textField.text
-
-  hasFocus: textField.activeFocus
+import "../components" as MMComponents
+import "../components/private" as MMPrivateComponents
 
-  content: TextField {
-    id: textField
+MMPrivateComponents.MMBaseSingleLineInput {
+  id: root
 
-    anchors.fill: parent
-    anchors.verticalCenter: parent.verticalCenter
+  textField.echoMode: eyeButton.pressed ? TextInput.Normal : TextInput.Password
 
-    color: root.enabled ? __style.nightColor : __style.mediumGreenColor
-    placeholderTextColor: __style.darkGreyColor
-    font: __style.p5
-    hoverEnabled: true
-    echoMode: eyeButton.pressed ? TextInput.Normal : TextInput.Password
-    inputMethodHints: Qt.ImhNoPredictiveText | Qt.ImhNoAutoUppercase
-    background: Rectangle {
-      color: __style.transparentColor
-    }
-  }
+  textField.inputMethodHints: textField.inputMethodHints | Qt.ImhNoAutoUppercase
```

Comment:
Won't this produce a binding loop? 🤔

Reply:
Hmm it does.. What's the proper way to do |= in qml?

Also why do we have `inputMethodHints: Qt.ImhNoPredictiveText` in the base class?

Reply:
We want to keep `inputMethodHints: Qt.ImhNoPredictiveText` for all inputs in the app for now. Predictive text was giving us a lot of issues.

I have not found a way to do `|=` in QML yet, but If you find a way, I will be excited! :)